library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -718,8 +765,114 @@ private void checkPermissions() {
     }
 
     @Override
-    public void showCurrentLocationBtn(boolean isVisible) {
-        currentLocationBtn.setVisibility(isVisible ? VISIBLE : GONE);
+    public void addFeaturePoints(FeatureCollection featureCollection) throws JSONException {
+        JSONArray featuresArray = this.featureCollectionJSON.getJSONArray("features");
+        for (com.mapbox.geojson.Feature feature : featureCollection.features()) {
+            String featureId = feature.id();
+            if (featureId != null && !featureMap.containsKey(featureId)) {
+                featureMap.put(featureId, featuresArray.length());
+                featuresArray.put(new JSONObject(feature.toJson()));
+            }
+        }
+        if (mapboxMap != null) {
+            ((GeoJsonSource) mapboxMap.getSource(primaryGeoJsonSource.getId())).setGeoJson(featureCollectionJSON.toString());
+        }
+    }
+
+    @Override
+    public void updateFeaturePointProperties(FeatureCollection featureCollection) throws JSONException {
```

Comment:
You cannot easily edit FeatureCollections.

If you get the List of Features in the FeatureCollection, there is no method to update the Feature. 

One would still need to create a new Feature object and I'm not sure that this is particularly any different since one would still use the `Feature.from` method to create a Feature from a json string or jsonobject.

Reply:
Unless maybe remove the property then add it.

Reply:
there is are several methods for manipulating a feature properties 
- `com.mapbox.geojson.Feature#removeProperty`
- `com.mapbox.geojson.Feature#addStringProperty`
- `com.mapbox.geojson.Feature#addProperty`

Reply:
Cool. Made the changed.